test/cppsuite/src/main/database.cpp

```diff
@@ -72,11 +102,27 @@ database::get_collection(uint64_t id)
 collection &
 database::get_random_collection()
 {
-    size_t collection_count = get_collection_count();
+    std::lock_guard<std::mutex> lg(_mtx);
+
     /* Any caller should expect at least one collection to exist. */
-    testutil_assert(collection_count != 0);
-    return (get_collection(
-      random_generator::instance().generate_integer<uint64_t>(0, collection_count - 1)));
+    testutil_assert(!_collections.empty());
+
+    auto it = _collections.begin();
+    std::advance(it, rand() % _collections.size());
+    return it->second;
+}
+
+std::string
+database::get_random_collection_name()
```

Comment:
I guess it could be something else to improve in the cpp suite framework... As you said this would impact other tests as we need the guarantee to retrieve a valid collection when using `get_random_collection`.
We could make `get_random_collection` return a null pointer when  `_collections` is empty and assert this is not null in all the existing tests?

Reply:
Yep that's what I was thinking. I was leaning towards returning `std::optional<collection>` rather than a null pointer though.